### PR TITLE
Technical / Branch docker_dev updated before merging to master

### PR DESCRIPTION
**Pull request by pawelirh**
Updated:
- README
- Dockerfile
- docker-compose.yaml
- workflow

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,49 +1,39 @@
 FROM ros:noetic-ros-core
```
Comment (Kotochleb) [inline Dockerfile:1]:
```suggestion
FROM ros:noetic-ros-base
```

According to [source dockerfile](https://github.com/osrf/docker_images/blob/master/ros/noetic/ubuntu/focal/ros-base/Dockerfile) of noetic-ros-base we wouldn't have to install rosdep separately with this base image.

**Review by Kotochleb — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,49 +1,32 @@
...
+WORKDIR /ros_ws
 
-# Python 3 dependencies
-RUN pip3 install \
-        rosdep \
-        rospkg
+# Update Ubuntu Software and Install components
+RUN curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash - && \
+    apt update && \
+    apt upgrade -y && \
```
Comment (Kotochleb) [inline Dockerfile:11]: Are we upgrading docker images or we are dropping this for future ones?

File: Dockerfile (hunk 1 above)
Comment (Kotochleb) [reply Dockerfile]: I see it was changes. I'm resolving this thread.

File: Dockerfile [hunk 3]
```diff
@@ -1,49 +1,39 @@
...
+    apt install -y \
+        curl \
+        git \ 
+        nodejs \
+        npm \
+        python3-dev \
+        python3-pip \
+        python3-rosdep \
+        python3-rospkg && \
```
[the remark this answers is not in the document]
Comment (Kotochleb) [reply Dockerfile]: I see it was changes. I'm resolving this thread.

File: Dockerfile (hunk 2 above)
Comment (pawelirh) [reply Dockerfile:11]: `apt upgrade` doesn't hurt us a lot (especially while comparing to nodejs build), I'd leave it as it is.

Comment (Kotochleb) [reply Dockerfile:11]: If you say so, then I'm resolving this thread.

**Pull request merged**